File: src/game/core/moves/ICaptureMove.java
package game.core.moves;

import game.core.Move;
import game.core.Square;

import java.util.List;

/**
 * Интерфейс для игр на которых фигура может захватывать фигуры противника.
 *
 * @author <a href="mailto:vladimir.romanov@gmail.com">Romanov V.Y.</a>
 */
public interface ICaptureMove extends Move {
    /**
     * Вернуть клетки на которых стоят захваченные фигуры.
     */
    List<Square> getCaptured();
}

File: GameCore/src/game/core/moves/ICaptureMove.kt
package game.core.moves

import game.core.Move
import game.core.Square

/**
 * Интерфейс для игр на которых фигура может захватывать фигуры противника.
 *
 * @author [Romanov V.Y.](mailto:vladimir.romanov@gmail.com)
 */
interface ICaptureMove : Move {
    /**
     * Вернуть клетки на которых стоят захваченные фигуры.
     */
    val captured: List<Square>
}